python/trezorlib/tests/device_tests/test_msg_recoverydevice_shamir.py

```diff
@@ -221,3 +221,140 @@ def test_abort(self):
         # check that the device is wiped
         features = self.client.call_raw(proto.Initialize())
         assert features.initialized is False
+
+    def test_warnings(self):
+        # 128 bits security, 3 of 6
+        mnemonics = [
+            "extra extend academic bishop cricket bundle tofu goat apart victim enlarge program behavior permit course armed jerky faint language modern",
+            "extra extend academic acne away best indicate impact square oasis prospect painting voting guest either argue username racism enemy eclipse",
+            "extra extend academic arcade born dive legal hush gross briefing talent drug much home firefly toxic analysis idea umbrella slice",
+        ]
+        word_count = len(mnemonics[0].split(" "))
+
+        ret = self.client.call_raw(
+            proto.RecoveryDevice(
+                passphrase_protection=False, pin_protection=False, label="label"
+            )
+        )
+
+        # Confirm Recovery
+        assert isinstance(ret, proto.ButtonRequest)
+        self.client.debug.press_yes()
+        ret = self.client.call_raw(proto.ButtonAck())
+
+        # Homescreen - consider aborting process
+        assert isinstance(ret, proto.ButtonRequest)
+        self.client.debug.press_no()
+        ret = self.client.call_raw(proto.ButtonAck())
+
+        # Homescreen - but then bail out in the warning
+        assert isinstance(ret, proto.ButtonRequest)
+        self.client.debug.press_no()
+        ret = self.client.call_raw(proto.ButtonAck())
+
+        # Homescreen - click Enter
+        assert isinstance(ret, proto.ButtonRequest)
+        self.client.debug.press_yes()
+        ret = self.client.call_raw(proto.ButtonAck())
+
+        # Enter word count
+        assert ret == proto.ButtonRequest(
+            code=proto.ButtonRequestType.MnemonicWordCount
+        )
+        self.client.debug.input(str(word_count))
+        ret = self.client.call_raw(proto.ButtonAck())
+
+        # Homescreen
+        assert isinstance(ret, proto.ButtonRequest)
+        self.client.debug.press_yes()
+        ret = self.client.call_raw(proto.ButtonAck())
+
+        # Enter first share
+        assert ret == proto.ButtonRequest(code=proto.ButtonRequestType.MnemonicInput)
+        self.client.transport.write(proto.ButtonAck())
+        for word in mnemonics[0].split(" "):
+            time.sleep(1)
+            self.client.debug.input(word)
+        ret = self.client.transport.read()
+
+        # Homescreen
+        assert isinstance(ret, proto.ButtonRequest)
+        self.client.debug.press_yes()
+        ret = self.client.call_raw(proto.ButtonAck())
+
+        for i in range(5):
+            assert ret == proto.ButtonRequest(
+                code=proto.ButtonRequestType.MnemonicInput
+            )
+            self.client.transport.write(proto.ButtonAck())
+            time.sleep(1)
+            if i == 0:
```

Comment:
I don't think it makes sense to have `for` here, since you're then using `if` conditions anyway. If you wanted to avoid repeating the five lines above, refactor it to a function.

Reply:
i made a separate test for first three words wrong, and a separate one for same share